tests/unit/syncModel/assetFolderTransformer.test.ts

```diff
@@ -0,0 +1,89 @@
+import { describe, expect, it } from "@jest/globals";
+import { AssetFolderContracts } from "@kontent-ai/management-sdk";
+
+import { transformAssetFolderModel } from "../../../src/modules/sync/modelTransfomers/assetFolder.js";
+import { AssetFolderSyncModel } from "../../../src/modules/sync/types/syncModel.js";
+
+describe("transformAssetFolderModel", () => {
+  it("correctly transforms asset folder model to sync model", () => {
+    const input = {
+      id: "01160b00-9d58-446d-bd13-5d9ceeb55d3c",
+      name: "folderName",
+      codename: "folderCodename",
+      external_id: "externalId",
+      folders: [
+        {
+          id: "folderId",
+          name: "nestedFolder",
+          codename: "nestedFolderCodename",
+          external_id: "nestedExternalId",
+          folders: [],
+        },
+      ],
+    } as const satisfies AssetFolderContracts.IAssetFolderContract;
+
+    const expectedOutput: AssetFolderSyncModel = {
+      name: "folderName",
+      codename: "folderCodename",
+      folders: [
+        {
+          name: "nestedFolder",
+          codename: "nestedFolderCodename",
+          folders: [],
+        },
+      ],
+    };
+
+    const result = transformAssetFolderModel(input);
+
+    expect(result).toStrictEqual(expectedOutput);
+  });
+
+  it("correctly transforms asset folder model to sync model with nested folders", () => {
```

Comment:
I think the tests are too similar to have two. I am not sure but I see nested folder in both. I would keep the second one :D Also I like that only the root folder has GUID, others weren't good enough for it :D